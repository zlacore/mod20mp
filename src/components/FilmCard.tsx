import type React from 'react';
import type Film from '../utils/interfaces/Film.interface';
import { IoEyeOutline } from 'react-icons/io5';
import { ImCross } from 'react-icons/im';
import { CgPlayListAdd } from 'react-icons/cg';

type FilmCardProps = {
  currentFilm: Film;
  addToWatchList?: (() => void) | null;
  addToSeenItList?: (() => void) | null;
  onWatchList?: boolean | null;
  onSeenItList?: boolean | null;
  removeFromStorage?:
    | ((
        e: React.MouseEvent<SVGSVGElement, MouseEvent>,
        currentlyOnWatchList: boolean | null | undefined,
        currentlyOnSeenItList: boolean | null | undefined,
        title: string | null
      ) => void)
    | null;
};

const FilmCard = ({
  currentFilm,
  addToWatchList,
  addToSeenItList,
  onWatchList,
  onSeenItList,
  removeFromStorage,
}: FilmCardProps) => {
  return (
    <>
      {currentFilm?.Title ? (
        <section className='filmCard'>
          <figure>
            <img src={`${currentFilm.Poster}`} alt={`${currentFilm.Title}`} />
          </figure>
          <article className='details'>
            <h2>{currentFilm.Title}</h2>
            <p>
              <strong>Directed By:</strong> {currentFilm.Director}
            </p>
            <p>
              <strong>Starring:</strong> {currentFilm.Actors}
            </p>
            <p>
              <strong>Released:</strong> {currentFilm.Released}
            </p>
            <p>
              <strong>Genre:</strong> {currentFilm.Genre}
            </p>
          </article>
          <article className='plot'>
            <p>
              <strong>Plot:</strong> {currentFilm.Plot}
            </p>
          </article>
          {onWatchList || onSeenItList ? (
            <aside className='icons'>
              <ImCross
                style={{ fontSize: '40px', cursor: 'pointer' }}
                onClick={(e: React.MouseEvent<SVGSVGElement, MouseEvent>) =>
                  removeFromStorage?.(
                    e,
                    onWatchList,
                    onSeenItList,
                    currentFilm.Title
                  )
                }
              />
            </aside>
          ) : (
            <aside className='icons'>
              <CgPlayListAdd
                style={{ fontSize: '50px', cursor: 'pointer' }}
                onClick={() => addToWatchList?.()}
              />
              <IoEyeOutline
                style={{ fontSize: '50px', cursor: 'pointer' }}
                onClick={() => addToSeenItList?.()}
              />
            </aside>
          )}
        </section>
      ) : (
        <h1 style={{ margin: '16px 0' }}>Please search for a film.</h1>
      )}
    </>
  );
};

export default FilmCard;
