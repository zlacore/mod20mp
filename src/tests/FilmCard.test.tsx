import pretty from 'pretty';
import { render } from '@testing-library/react';
import FilmCard from '../components/FilmCard';
import type Film from '../utils/interfaces/Film.interface';

const film: Film = {
  Title: 'Gamera, the Guardian of the Universe',
  Director: 'ShÃ»suke Kaneko',
  Actors: 'Tsuyoshi Ihara, Akira Onodera, Shinobu Nakayama',
  Released: '16 Apr 1997',
  Poster: 'https://m.media-amazon.com/images/M/MV5BNTFkNzAxYTUtZGE4ZC00NjBkLWE3ZDAtYjE4NzQ0MDBhZWQwXkEyXkFqcGdeQXVyNjQ2MzU1NzQ@._V1_SX300.jpg',
  Genre: 'Action, Drama, Fantasy',
  Plot: 'An ornithologist investigates reports of a monstrous new species of bird just as a teenage girl is gifted an amulet found on mysterious atoll. As the creatures begin to attack, an ancient guardian with a bond to the girl emerges.',
};

describe('FilmCard', () => {
  it('should contain the film information', () => {

    render(<FilmCard currentFilm={film}/>);

    const filmImg = document.querySelector('img');

    if (filmImg) {
      expect(filmImg.getAttribute('src')).toBe(
        'https://m.media-amazon.com/images/M/MV5BNTFkNzAxYTUtZGE4ZC00NjBkLWE3ZDAtYjE4NzQ0MDBhZWQwXkEyXkFqcGdeQXVyNjQ2MzU1NzQ@._V1_SX300.jpg'
      );  
    }
    
  });

  it('should match snapshot', () => {
    render(<FilmCard currentFilm={film} addToWatchList={null} addToSeenItList={null}/>);

    const filmContainer = document.querySelector('.filmCard') as HTMLElement;

    if (filmContainer) {
      expect(pretty(filmContainer.innerHTML)).toMatchSnapshot();
    }
  })
  
});
