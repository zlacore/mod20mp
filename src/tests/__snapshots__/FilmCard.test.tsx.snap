// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilmCard > should match snapshot 1`] = `
"<figure><img alt="Gamera, the Guardian of the Universe" src="https://m.media-amazon.com/images/M/MV5BNTFkNzAxYTUtZGE4ZC00NjBkLWE3ZDAtYjE4NzQ0MDBhZWQwXkEyXkFqcGdeQXVyNjQ2MzU1NzQ@._V1_SX300.jpg"></figure>
<article class="details">
  <h2>Gamera, the Guardian of the Universe</h2>
  <p><strong>Directed By:</strong> Sh√ªsuke Kaneko</p>
  <p><strong>Starring:</strong> Tsuyoshi Ihara, Akira Onodera, Shinobu Nakayama</p>
  <p><strong>Released:</strong> 16 Apr 1997</p>
  <p><strong>Genre:</strong> Action, Drama, Fantasy</p>
</article>
<article class="plot">
  <p><strong>Plot:</strong> An ornithologist investigates reports of a monstrous new species of bird just as a teenage girl is gifted an amulet found on mysterious atoll. As the creatures begin to attack, an ancient guardian with a bond to the girl emerges.</p>
</article>
<aside class="icons"><svg stroke="currentColor" fill="none" stroke-width="0" viewBox="0 0 24 24" style="font-size: 50px; cursor: pointer;" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg">
    <path d="M2 5H14V7H2V5Z" fill="currentColor"></path>
    <path d="M2 9H14V11H2V9Z" fill="currentColor"></path>
    <path d="M10 13H2V15H10V13Z" fill="currentColor"></path>
    <path d="M16 9H18V13H22V15H18V19H16V15H12V13H16V9Z" fill="currentColor"></path>
  </svg><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" style="font-size: 50px; cursor: pointer;" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg">
    <path fill="none" stroke-linecap="round" stroke-linejoin="round" stroke-width="32" d="M255.66 112c-77.94 0-157.89 45.11-220.83 135.33a16 16 0 0 0-.27 17.77C82.92 340.8 161.8 400 255.66 400c92.84 0 173.34-59.38 221.79-135.25a16.14 16.14 0 0 0 0-17.47C428.89 172.28 347.8 112 255.66 112z"></path>
    <circle cx="256" cy="256" r="80" fill="none" stroke-miterlimit="10" stroke-width="32"></circle>
  </svg></aside>"
`;
